src/wpool/mongoose_wpool.erl

```diff
@@ -121,8 +113,12 @@ start_configured_pools(PoolsIn) ->
     start_configured_pools(PoolsIn, ?ALL_HOST_TYPES).
 
 start_configured_pools(PoolsIn, HostTypes) ->
+    HostTypeSpecific = get_host_type_specific_pools(HostTypes),
+    start_configured_pools(PoolsIn, HostTypeSpecific, HostTypes).
+
+start_configured_pools(PoolsIn, HostTypeSpecific, HostTypes) ->
+    Pools = expand_pools(PoolsIn, HostTypeSpecific, HostTypes),
     [call_callback(init, PoolType, []) || PoolType <- get_unique_types(PoolsIn)],
```

Comment:
Why is init not called for host-type-specific pools? They don't need to appear on the top level at all.